lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -772,6 +827,32 @@ uint WebAssemblyModule::GetGlobalsByteSize() const
     return size;
 }
 
+void
+WebAssemblyModule::AddCustomSection(Wasm::CustomSection customSection)
+{
+    if (!m_customSections)
+    {
+        m_customSections = Anew(&m_alloc, CustomSectionsList, &m_alloc);
+    }
+    m_customSections->Add(customSection);
+}
+
+uint32
+WebAssemblyModule::GetCustomSectionCount() const
+{
+    return m_customSections ? (uint32)m_customSections->Count() : 0;
+}
+
+Wasm::CustomSection
+WebAssemblyModule::GetCustomSection(uint32 index) const
+{
+    if (index >= GetCustomSectionCount())
+    {
+        throw Wasm::WasmCompilationException(_u("Custom section index out of bounds %u"), index);
```

Comment:
It's true that it shouldn't happen, but lets say this api is used differently and we pass an out of bounds index, it should throw an exception we control. From what I can see we have the same kind of pattern in `WebAssemblyModule::EntryImports`.
Either we 
- don't use this api in `EntryImports` and `EntryCustomSections`
- Try/catch for `WasmCompilationException`
- remove the check
What do you think ?